.github/workflows/docs.yml

```diff
@@ -0,0 +1,51 @@
+name: Documentation
+on:
+  pull_request:
+    paths:
+      - .vale.ini
+      - docs/**
+
+permissions:
+  contents: read
+
+jobs:
+  docs:
+    if: github.repository == 'Homebrew/brew'
+    runs-on: ubuntu-22.04
+    defaults:
+      run:
+        working-directory: docs
+    steps:
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - name: Run vale for docs linting
+        run: |
+          brew install vale
+          vale .
+
+      - name: Install Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "2.7"
+          bundler-cache: true
+          working-directory: docs
+
+      - name: Build docs site
+        run: bundle exec rake build
+
+#      - name: Check Markdown syntax (currently failing)
+#        run: bundle exec rake lint
+
+      - name: Checkout Homebrew/rubydoc.brew.sh
+        uses: actions/checkout@main
+        with:
+          repository: Homebrew/rubydoc.brew.sh
+          path: rubydoc.brew.sh
+
+      - name: Run yard for rubydoc output
+        working-directory: rubydoc.brew.sh
```

Comment:
If you're having issues with the working directory (I'm guessing having `.yardopt` outside of the working directory is causing issues), you could set the working directory to `Library/Homebrew` and set the `BUNDLE_GEMFILE` environment variable to `rubydoc.brew.sh/Gemfile`.